From 967e79b7cfaacfdf463b865b33433029aff0f9d6 Mon Sep 17 00:00:00 2001
From: Chao Zeng <chao.zeng@siemens.com>
Date: Thu, 27 May 2021 10:09:04 +0800
Subject: [PATCH 8/9] sf: query the write protect status before operating the
 device

We need to query the write protect status before operating the device,
and if not permit, return a prompt.

Signed-off-by: Chao Zeng <chao.zeng@siemens.com>
---
 drivers/mtd/spi/sf_probe.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mtd/spi/sf_probe.c b/drivers/mtd/spi/sf_probe.c
index 6c87434867..4224439ee8 100644
--- a/drivers/mtd/spi/sf_probe.c
+++ b/drivers/mtd/spi/sf_probe.c
@@ -109,6 +109,11 @@ static int spi_flash_std_write(struct udevice *dev, u32 offset, size_t len,
 	struct mtd_info *mtd = &flash->mtd;
 	size_t retlen;
 
+	if(flash->flash_is_locked && flash->flash_is_locked(flash,offset,len)) {
+		debug("SF: Write protection is enable\n");
+		return -ENOPROTOOPT;
+	}
+
 	return mtd->_write(mtd, offset, len, &retlen, buf);
 }
 
@@ -127,6 +132,11 @@ static int spi_flash_std_erase(struct udevice *dev, u32 offset, size_t len)
 	instr.addr = offset;
 	instr.len = len;
 
+	if(flash->flash_is_locked && flash->flash_is_locked(flash,offset,len)) {
+		debug("SF: Write protection is enable\n");
+		return -ENOPROTOOPT;
+	}	
+
 	return mtd->_erase(mtd, &instr);
 }
 
-- 
2.32.0

